pkg/bpf/map_linux_test.go

```diff
@@ -87,16 +113,270 @@ func (s *BPFPrivilegedTestSuite) TestOpen(c *C) {
+	isNew, err := existingMap.OpenOrCreate()
+	c.Check(err, IsNil) // Avoid assert to ensure Close() is called below.
+	c.Check(isNew, Equals, false)
+
+	// preallocMap unsets BPF_F_NO_PREALLOC. OpenOrCreate should recreate map.
+	EnableMapPreAllocation() // prealloc on/off is controllable in HASH map case.
+	preallocMap := NewMap("cilium_test",
+		MapTypeHash,
+		int(unsafe.Sizeof(TestKey{})),
+		int(unsafe.Sizeof(TestValue{})),
+		maxEntries,
+		0,
+		0,
+		DumpParserFunc).WithCache()
+	// not to mess up emitted test results, checking isNew is sufficient.
+	logLevel := logging.DefaultLogger.GetLevel()
+	logging.DefaultLogger.SetLevel(logrus.ErrorLevel)
+	isNew, err = preallocMap.OpenOrCreate()
+	c.Check(err, IsNil)
+	c.Check(isNew, Equals, true)
+	logging.DefaultLogger.SetLevel(logLevel)
+	DisableMapPreAllocation()
+
+	// preallocMap is already open. OpenOrCreate does nothing.
+	isNew, err = preallocMap.OpenOrCreate()
+	c.Check(err, IsNil)
+	c.Check(isNew, Equals, false)
+
+	err = preallocMap.Close()
+	c.Assert(err, IsNil)
+}
+
+func (s *BPFPrivilegedTestSuite) TestOpenParallel(c *C) {
+	parallelMap := NewMap("cilium_test",
+		MapTypeHash,
+		int(unsafe.Sizeof(TestKey{})),
+		int(unsafe.Sizeof(TestValue{})),
+		maxEntries,
+		BPF_F_NO_PREALLOC,
+		0,
+		DumpParserFunc).WithCache()
+	isNew, err := parallelMap.OpenParallel()
+	c.Check(isNew, Equals, true)
+
+	isNew, err = parallelMap.OpenParallel()
+	c.Check(isNew, Equals, false)
+	c.Check(err, Not(IsNil))
+
+	noDiff := parallelMap.DeepEquals(testMap)
+	c.Check(noDiff, Equals, true)
+
+	key1 := &TestKey{Key: 101}
+	value1 := &TestValue{Value: 201}
+	key2 := &TestKey{Key: 102}
+	value2 := &TestValue{Value: 202}
+
+	err = testMap.Update(key1, value1)
+	c.Check(err, IsNil)
+	err = parallelMap.Update(key2, value2)
+	c.Check(err, IsNil)
+
+	value, err := testMap.Lookup(key1)
+	c.Check(err, IsNil)
+	c.Check(value, DeepEquals, value1)
+	value, err = testMap.Lookup(key2)
+	c.Check(err, Not(IsNil))
+	c.Check(value, IsNil)
+
+	value, err = parallelMap.Lookup(key1)
+	c.Check(err, Not(IsNil))
+	c.Check(value, IsNil)
+	value, err = parallelMap.Lookup(key2)
+	c.Check(err, IsNil)
+	c.Check(value, DeepEquals, value2)
+
+	err = parallelMap.EndParallelMode()
+	c.Check(err, IsNil)
+	err = parallelMap.EndParallelMode()
+	c.Check(err, Not(IsNil))
+}
+
+func (s *BPFPrivilegedTestSuite) TestBasicManipulation(c *C) {
+	// existingMap is the same as testMap. Opening should succeed.
+	existingMap := NewMap("cilium_test",
+		MapTypeHash,
+		int(unsafe.Sizeof(TestKey{})),
+		int(unsafe.Sizeof(TestValue{})),
+		maxEntries,
+		BPF_F_NO_PREALLOC,
+		0,
+		DumpParserFunc).WithCache()
+	err := existingMap.Open()
+	c.Check(err, IsNil) // Avoid assert to ensure Close() is called below.
+
+	key1 := &TestKey{Key: 103}
+	value1 := &TestValue{Value: 203}
+	key2 := &TestKey{Key: 104}
+	value2 := &TestValue{Value: 204}
+
+	err = existingMap.Update(key1, value1)
+	c.Check(err, IsNil)
+	// key    val
+	// 100    200
```

Comment:
Are these comments correct? `key` is defined above with the value 103, and the value is different too. Same for the similar comments below in the test.

Reply:
Simple mistake, thanks, I'll fix this.

Reply:
Done (https://github.com/cilium/cilium/pull/7538/commits/8280912414b4b50800a879c2582e80a4fe08eb54)